### ISSUE TITLE
`react/no-unused-class-component-methods` false positive when method is referenced via a ref

### ISSUE THREAD

**Issue by jcrosetto** [jsx-eslint/eslint-plugin-react#3374]
The `react/no-unused-class-component-methods` rule will incorrectly flag a method as unused if it is only referenced through a ref.

A minimal example:
```
class MyComponent {
...
public refMethod() { } // Linter says this is unused
...
}

function MyOtherComponent {
  const myComponentRef = useRef();

  useEffect(() => {
    myComponentRef.current?.refMethod();
  }, []);  

  return <MyComponent ref={myComponentRef} />;
}

**Comment by ljharb** [comment]
React components are not supposed to be used like a class instance, and I’m surprised a ref gives you the actual component instance.

At any rate, JS isn’t a typed language, and static analysis has limits, so this is simply not something a linter can know. I’d suggest using an eslint override comment.

**Comment by jcrosetto** [comment]
I had the same reaction that refs could be used to access functions on the class.

I figured this would probably be beyond the capabilities of the linter but wanted to raise it just in case. I'll take the route of disabling the linter on that line. Thanks!
